types/options.d.ts

```diff
@@ -66,7 +66,7 @@ export interface ComponentOptions<
   renderError?: (h: () => VNode, err: Error) => VNode;
   staticRenderFns?: ((createElement: CreateElement) => VNode)[];
 
-  beforeCreate?(): void;
+  beforeCreate?(this: Vue): void;
```

Comment:
I think ThisType of `beforeCreate` should be extended component type because there are cases that we want to intialize additional data in it.